### PR TITLE
Protection Domain resource CRUD and Import

### PR DESCRIPTION
**Pull request by rounak-adhikary**
image

### CODE REVIEW

**Comment by rounak-adhikary** [comment]
image

File: powerflex/protection_domain_resource.go [hunk 1]
```diff
@@ -0,0 +1,579 @@
...
+}
+
+func (d *protectionDomainResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	d.client = req.ProviderData.(*goscaleio.Client)
+
+	systems, err := d.client.GetSystems()
```
Comment (shenda1) [inline powerflex/protection_domain_resource.go:159]: can we use getFirstSystem() here?

File: go.mod [hunk 2]
```diff
@@ -3,7 +3,7 @@ module terraform-provider-powerflex
 go 1.19
 
 require (
-	github.com/dell/goscaleio v1.10.1-0.20230425105152-3468c1df396b
+	github.com/dell/goscaleio v1.10.1-0.20230426073125-2b0de5cb310f
 	github.com/hashicorp/terraform-plugin-docs v0.13.0
 	github.com/hashicorp/terraform-plugin-framework v1.0.1
```
Comment (shenda1) [inline go.mod:8]: update the version to 1.1.1

File: powerflex/protection_domain_resource_schema.go [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+				boolplanmodifier.UseStateForUnknown(),
+			},
+		},
+		"state": schema.StringAttribute{
+			Description:         "State of the PD.",
+			MarkdownDescription: "State of the PD.",
+			Computed:            true,
+		},
+		"rf_cache_accp_id": schema.StringAttribute{
+			Description:         "ID of the Rf Cache Acceleration Pool attached to the PD.",
```
Comment (shenda1) [inline powerflex/protection_domain_resource_schema.go:48]: RF

File: powerflex/protection_domain_resource.go [hunk 4]
```diff
@@ -0,0 +1,579 @@
...
+		return
+	} else if numSys > 1 {
+		resp.Diagnostics.AddError("More than one system found", "")
+		return
+	}
+	system, err := d.client.FindSystem(systems[0].ID, "", "")
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to Read Powerflex Protection Domains",
+			err.Error(),
```
Comment (shenda1) [inline powerflex/protection_domain_resource.go:178]: protection domain -> system?

Comment (rounak-adhikary) [reply powerflex/protection_domain_resource.go:159]: done

File: powerflex/protection_domain_resource_schema.go (hunk 3 above)
Comment (rounak-adhikary) [reply powerflex/protection_domain_resource_schema.go]: done

File: go.mod (hunk 2 above)
Comment (rounak-adhikary) [reply go.mod]: udgraded

File: powerflex/protection_domain_resource.go (hunk 1 above)
Comment (rounak-adhikary) [reply powerflex/protection_domain_resource.go]: done

File: powerflex/protection_domain_resource.go [hunk 5]
```diff
@@ -0,0 +1,560 @@
...
+func (d *protectionDomainResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	// Retrieve values from state
+	var state protectionDomainResourceModel
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// id := state.ID.ValueString()
```
Comment (shenda1) [inline powerflex/protection_domain_resource.go:303]: Please remove comments if not required. Applicable for all such instances.

Comment (rounak-adhikary) [reply powerflex/protection_domain_resource.go]: done. Pls check.

**Review by shenda1 — APPROVED** [review]

**Review by Krishnan-Priyanshu — APPROVED** [review]
LGTM

**Pull request merged**